docs/DeveloperGuide.md

```diff
@@ -2,37 +2,220 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+Referred to https://se-education.org/addressbook-level3/DeveloperGuide.html when drafting the DG
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
 
+![](./images/architeture.png)
+
+The Architecture Diagram given above explains the high-level design of the App.
+
+The **User** sends command which is processed by Input
+
+**Input**: Receives user input string, processes it and sends to CashLeh main app
+
+**CashLeh**: main app
+
+**UI**: Responsible for the printing onto CLI
+
+**Parser**: Process the input string and calls respective methods
+
+**FileStorage**: Reads data from, and writes data to file
+
+### How the architecture components interact with each other
+
+![](./images/main_example.png)
+
+The Sequence Diagram above shows how the components interact with each other when the user enters a command
+
+### Setting, viewing and editing a budget feature
+
+When CashLeh? is launched, after asking for the user's name, the app will ask whether the user would like to set a
+budget to manage his/her transactions. CashLeh? will still work seamlessly, even if no budget is set, but some functions
+(such as viewing your budget) will obviously not be available for use.
+
+The option to set, view and edit a budget is managed by the `BudgetHandler`. An object of this class is constructed
+with two attributes, of type `FinancialStatement` and `Budget` respectively. It features both a setter and a getter
+which will be needed for the functioning of the overall budget feature. The BudgetHandler handles a budget which is
+initially set to not active (activated only when the budget is actually set by the user through the command
+`updateBudget`). On the other hand, the BudgetHandler is simultaneously responsible for measuring the Progress: It
+sets the Progress' percentage and calls the method to display the progress bar chart. Below you can see a simple class
+diagram summarizing these relationships.
+
+![](./images/budgetHandler_classDiagram.png)
+
+When the user enters the command to view a budget `viewBudget`, the following happens:
+
+![](./images/budgetHandler_sequenceDiagram.png)
```

Comment:
These two images aren't appearing in the developer's guide on my machine. I am running Safari on MacOS. Not sure if this issue is present on other devices as well.

<img width="1018" alt="image" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/97012075/05724c85-0bb6-4a32-8084-7df49995786a">

But yes, in general the sequence diagram contains similar issues as what @lctxct and @bnjm2000 have pointed out above.